src/plugins/data/server/index_patterns/routes/update_index_pattern.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { IRouter } from '../../../../../core/server';
+import { assertIndexPatternsContext } from './util/assert_index_patterns_context';
+import { handleErrors } from './util/handle_errors';
+import { fieldSpecSchema, serializedFieldFormatSchema } from './util/schemas';
+
+const indexPatternUpdateSchema = schema.object({
+  title: schema.maybe(schema.string()),
+  type: schema.maybe(schema.string()),
+  typeMeta: schema.maybe(schema.object({}, { unknowns: 'allow' })),
+  timeFieldName: schema.maybe(schema.string()),
+  intervalName: schema.maybe(schema.string()),
+  sourceFilters: schema.maybe(
+    schema.arrayOf(
+      schema.object({
+        value: schema.string(),
+      })
+    )
+  ),
+  fieldFormats: schema.maybe(schema.recordOf(schema.string(), serializedFieldFormatSchema)),
+  fields: schema.maybe(schema.recordOf(schema.string(), fieldSpecSchema)),
+});
+
+export const registerUpdateIndexPatternRoute = (router: IRouter) => {
+  router.post(
+    {
+      path: '/api/index_patterns/index_pattern/{id}',
+      validate: {
+        params: schema.object(
+          {
+            id: schema.string({
+              minLength: 1,
+              maxLength: 1_000,
+            }),
+          },
+          { unknowns: 'allow' }
+        ),
+        body: schema.object({
+          skip_field_refresh: schema.maybe(schema.boolean({ defaultValue: false })),
```

Comment:
There's a number of places that accept 'skip_refresh_fields' - I think it would make more sense to default to 'true' or rename it 'refresh_fields'.

For the most part, this api will be used to change behavior in kibana. I don't think there will be much use of the resulting index pattern spec.

Reply:
Done.